doc/ref_arch/openstack/chapters/chapter04.md

```diff
@@ -1,82 +1,209 @@
+
+As we get away from the large data centers to the smaller sites it becomes progressively difficult to be able to create enough capacity for each of these instance types in support of their target VNFs or to have a mix of hardware targeted for each instance type. 
+
+<a name="4.3.3"></a>
+### 4.3.3. Containerised OpenStack Services
+
+#### 4.3.3.1. Architectural Drivers – Requirements Traceability
+
+| Ref # | sub-category | Description |
+|----|----|-----|
+| req.gen.cnt.02 | Cloud nativeness | The Architecture should consist of service components implemented as microservices that are individually dynamically scalable. |
+| req.vim.02 | General | The Architecture should support deployment of OpenStack components in containers. |
+| req.gen.rsl.01 | Resiliency | The Architecture must support resilient OpenStack components that are required for the continued availability of running workloads. |
+
+#### 4.3.3.2. Justification
+
+ontainers are lightweight compared to Virtual Machines and leads to efficient resource utilization. Kubernetes auto manages scaling, recovery from failures, etc.  Thus, it is recommended that the OpenStack services be containerized for resiliency and resource efficiency.
 
 
 <a name="4.4"></a>
-## 4.4 Cloud Controller Services
-- Core Components (keystone, heat, nova, neutron, glance)
-- Identity Management
+## 4.4 Compute Node Services
 
+In Section 3.5 the high-level set of infrastructure components needed to run VMs was presented.  This section is primarily about scheduling the VM to be created onto a particular physical server. This section delves into the compute host selection for these VMs, the resource pools, hardware and software considerations.
 
-<a name="4.5"></a>
-## 4.5 Cloud Workload Services.
+<p align="center"><img src="../figures/Figure_4_3_Select_host_instance_launch.png" alt="Selection of a compute host to launch an instance"></br>Figure 4-3. Selection of a compute host to launch an instance.</p>
+
+When a request for an instance creation is made, the requested features and capabilities of the instance is used to determine the host on which the instance should be launched (Figure 4-3).  The nova scheduler service matches requested features and capabilities of the instance and the capabilities/configuration of the hosts (hardware and software).  The nova-scheduler can be configured to use the host aggregates (and availability zones) in selecting the hosts when an instance create request with these capabilities/configurations is requested. The Common Telco NFVI has only a handful of standard profiles and hence the number of potential target host servers (Figure 4-3) may be large if only the host-aggregates filter is utilized and, thus, other filters should also be selected; affinity/non-affinity, custom key-value pairs, etc. 
```

Comment:
I think this description is irrelevant here. This chapter should decribe what runs on an compute host and not how it is selected.

Reply:
i guess we need to only say: use host aggregate to differentiate between various profiles. Again, let us approve it now and submit a new PR to clarify and simolify. Not Ideal, but it is important to get something out there for Antwrep.

Reply:
I agree, but it should say it instead of the description of Nova inner mechanics. 
I've added this to #339